engine/apps/slack/tasks.py

```diff
@@ -771,3 +771,29 @@ def clean_slack_integration_leftovers(organization_id, *args, **kwargs):
     OnCallSchedule.objects.filter(organization_id=organization_id).update(channel=None)
     logger.info(f"Cleaned OnCallSchedule slack_channel_id for organization {organization_id}")
     logger.info(f"Finish clean slack leftovers for organization {organization_id}")
+
+
+@shared_dedicated_queue_retry_task(autoretry_for=(Exception,), retry_backoff=True, max_retries=10)
+def clean_slack_channel_leftovers(slack_team_identity_id, slack_channel_id):
+    """
+    This task removes binding to slack channel after channel arcived or deleted in slack.
+    """
+    SlackTeamIdentity = apps.get_model("slack", "SlackTeamIdentity")
+    ChannelFilter = apps.get_model("alerts", "ChannelFilter")
+
+    try:
+        sti = SlackTeamIdentity.objects.get(id=slack_team_identity_id)
+    except SlackTeamIdentity.DoesNotExist:
+        logger.info(
+            f"Failed to clean_slack_channel_leftovers slack_channel_id={slack_channel_id} slack_team_identity_id={slack_team_identity_id} : Invalid slack_team_identity_id"
+        )
+        return
+
+    for org in sti.organizations.all():
```

Comment:
I would filter orgs by `general_log_channel_id` here instead of taking all. I would also use bulk update for them

Reply:
I can't filter orgs by general log channel here, I need to clean ChannelFilters in all organizations where slack was installed

Reply:
Anyway, I introduced bulk update of orgs and found fixes issue with updating of channel filters